### ISSUE TITLE
How can we fix the moving_var and moving_mean in the batch norm layer.

### ISSUE THREAD

**Issue by horserma** [apache/incubator-mxnet#1536]
I am trying to use Residual Network and one thing is confusing. For the batch norm layer, I only saw there is a arg to fixe gamma during training. I am wondering if it is possible to have options to fixed moving_var and moving_mean, since Kaiming He expained in his page that the moving_var and mean  are calculated by a large scale dataset, and changing then can lead to different results. I mean, to make things more flexible, is it possible to freeze them as well?

**Comment by horserma** [comment]
It this possible? I tried to change the momentum of batch norm layer to 1. it seems the parameters are still changing. Any way to freeze the parameters? @tqchen @piiswrong

**Comment by ascust** [comment]
In Kaiming He's implementation, it seems they have a parameter called "use_global_stats". Do we have such a thing? It means we dont calculate the mean from the batch but use the global fixed one instead. @tqchen

**Comment by horserma** [comment]
This is what I am thinking about. When fine tuning a model, If the batch is so small during training, the batch mean and variance may not be sufficient enough. It might be better to just use global variance instead of calculating the batch mean and variance each time. What do you think? @tqchen @piiswrong

**Comment by tqchen** [comment]
I see what you mean, this could be an possible option added to the BatchNorm, you are welcomed to open an PR

**Comment by horserma** [comment]
I have never down this before, so could you give me some hints? Is that just modifying the c++ code? What is the general procedure of doing this? @tqchen

**Comment by ascust** [comment]
@tqchen @piiswrong  I have modified the "batch_norm-inl.h" file, but I am wondering if I have to modify the "cudnn_batch_norm-inl.h". I found that even I compile mxnet with use_cudnn = 1. I am still using the "batch_norm-inl.h" except I explicitly define my symbol as
```
bn = mx.symbol.CuDNNBatchNorm(data=data, name="bn")
```
Is this supposed to work like this? I thought I was like after compiling with "use_cudnn = 1". The program will call Cudnn version automatically even I use "mx.symbol.BatchNorm" to define my symbol. Could you explain to me a little bit how this works and I will send a PR later. Thanks.

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
